lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -570,30 +570,47 @@ sub prepare_job_results {
     my %descriptions = map { $_->name => $_->description } @descriptions;
 
     my @wanted_jobs = grep {
-            (not $states         or $states->{$_->state})
-        and (not $results        or $results->{$_->result})
-        and (not $archs          or $archs->{$_->ARCH})
-        and (not $machines       or $machines->{$_->MACHINE})
-        and (not $failed_modules or $_->result eq OpenQA::Jobs::Constants::FAILED)
+              (not $states         or $states->{$_->state})
+          and (not $results        or $results->{$_->result})
+          and (not $archs          or $archs->{$_->ARCH})
+          and (not $machines       or $machines->{$_->MACHINE})
+          and (not $failed_modules or $_->result eq OpenQA::Jobs::Constants::FAILED)
     } @$jobs;
-    my @jobids = map { $_->id } @wanted_jobs;
+    my @jobids                = map { $_->id } @wanted_jobs;
```

Comment:
It is likely possible to combine the `grep` and `map` (by mapping unwanted jobs to an empty list).

Reply:
Yes, it would be possible, but we still need the `@wanted_jobs` array later

Reply:
Ok, then it makes sense to keep it separate.